### DISCUSSION TITLE
客户端是否有可能加入一个常用VPN协议的inbound代替Socks5？模拟L2TP或者PPTP之类的。

### DISCUSSION THREAD

**Discussion by Kylejustknows** [v2fly/v2ray-core#1191] [Ideas]
现有的客户连入方式，差不多只有Socks5了。有很多局限性。

如果V2Ray客户端能模拟一个“VPN协议服务器”（VPN连接后, V2ray客户端内部虚拟一个网关转发所有流量）就能：

1，在不折腾本地网络路由的基础上，支持局域网内更广泛的各种设备接入
2，转发TCP, UDP外更多的网络协议（ICMP 协议等）
3，Windows电脑对着自己建立VPN连接，然后把“代理服务器”的IP静态route成跳过VPN路由直连，Windows电脑就能不使用任何第三方软件全局代理。

在原理上是否可行？

**Comment by okudayukiko** [comment]
最簡單的方法是用TAP Windows。
